### PR TITLE
feat: add should forward props to box and text

### PR DESCRIPTION
**Pull request by SiTaggart**
Taking over from https://github.com/twilio-labs/paste/pull/2067 with some extra additions for custom style props that we add to the system.

### CODE REVIEW

**Review by zahnster — APPROVED** [review]
👍🏼  This looks great, much cleaner html on the implementation side.

File: packages/paste-core/components/modal/__tests__/index.spec.tsx [hunk 1]
```diff
@@ -156,7 +156,7 @@ describe('Modal', () => {
     expect(document.activeElement).toEqual(getByTestId('modal-header').querySelector('button'));
   });
 
-  it('should focus on the element provided as the initialFocus element in the modal', () => {
+  it.skip('should focus on the element provided as the initialFocus element in the modal', () => {
```
Comment (SiTaggart) [inline packages/paste-core/components/modal/__tests__/index.spec.tsx:159]: This test causes a blow up for some reason now. totally can't figure out what is up with it.

Comment (shleewhite) [reply packages/paste-core/components/modal/__tests__/index.spec.tsx:159]: will you investigate that in another PR?

**Review by shleewhite — APPROVED** [review]

File: packages/paste-website/src/components/shortcodes/table-of-contents/TableOfContentsListItem.tsx [hunk 2]
```diff
@@ -1,21 +1,21 @@
 import {styled, themeGet} from '@twilio-paste/styling-library';
 import type {TableOfContentsListItemProps} from './types';
 
-export const marginTopPartial = ({depth}: TableOfContentsListItemProps): string => {
+export const marginTopPartial = ({depth}: TableOfContentsListItemProps): any => {
```
Comment (SiTaggart) [inline packages/paste-website/src/components/shortcodes/table-of-contents/TableOfContentsListItem.tsx:4]: By adding the types package themeGet is retyped as returning `any` which is kind of annoying, but I get it.

File: packages/paste-core/components/modal/__tests__/index.spec.tsx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-core/components/modal/__tests__/index.spec.tsx:159]: Whoa, actual bug, terrible message. Good catch Jest

Comment (SiTaggart) [reply packages/paste-core/components/modal/__tests__/index.spec.tsx]: Fixed

**Review by zahnster — APPROVED** [review]
👍🏼

**Review by shleewhite — APPROVED** [review]

**Pull request merged**